src/contract/sent_transaction.ts

```diff
@@ -53,61 +49,38 @@ export class SentTransaction<T> {
   };
 
   constructor(
-    public signTransaction: ClientOptions["signTransaction"],
-    public assembled: AssembledTransaction<T>,
+    public options: SentTransactionOptions<T>,
+    public signed: Tx,
   ) {
-    if (!signTransaction) {
-      throw new Error(
-        "You must provide a `signTransaction` function to send a transaction",
-      );
-    }
-    this.server = new Server(this.assembled.options.rpcUrl, {
-      allowHttp: this.assembled.options.allowHttp ?? false,
+    this.server = new Server(this.options.rpcUrl, {
+      allowHttp: this.options.allowHttp ?? false,
     });
   }
 
   /**
-   * Initialize a `SentTransaction` from an existing `AssembledTransaction` and
-   * a `signTransaction` function. This will also send the transaction to the
-   * network.
+   * Initialize a `SentTransaction` from `options` and a `signed` 
+   * AssembledTransaction. This will also send the transaction to the network.
    */
   static init = async <U>(
-    /** More info in {@link MethodOptions} */
-    signTransaction: ClientOptions["signTransaction"],
-    /** {@link AssembledTransaction} from which this SentTransaction was initialized */
-    assembled: AssembledTransaction<U>,
+    /**
+     * Configuration options for initializing the SentTransaction.
+     * 
+     * @typedef {Object} SentTransactionOptions
+     * @property {number} [timeoutInSeconds] - Optional timeout duration in seconds for the transaction.
+     * @property {string} rpcUrl - The RPC URL of the network to which the transaction will be sent.
+     * @property {boolean} [allowHttp] - Optional flag to allow HTTP connections (default is false, HTTPS is preferred).
+     * @property {(xdr: xdr.ScVal) => U} parseResultXdr - Function to parse the XDR result returned by the network.
```

Comment:
Reminder: these [don't show up anywhere](https://github.com/stellar/js-stellar-sdk/pull/962#discussion_r1596950321)!

Reply:
Hopefully they show up in the generated html docs??